File: src-tauri/src/models/history.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct HistoryEntry {
    pub manga_id: String,
    pub manga_title: String,
    pub manga_path: String,
    pub manga_cover_path: String,
    pub manga_chapter_count: usize,
    pub chapter_id: String,
    pub chapter_title: String,
    pub chapter_path: String,
    pub chapter_file_type: String, // "dir" | "cbz"
    pub last_read_at: u64,         // unix epoch seconds
}

#[derive(Debug, Clone, Default, Serialize, Deserialize)]
pub struct HistoryData {
    pub entries: Vec<HistoryEntry>, // newest first, deduped by chapter_id
}

File: src/types/history.ts
export interface HistoryEntry {
  manga_id: string;
  manga_title: string;
  manga_path: string;
  manga_cover_path: string;
  manga_chapter_count: number;
  chapter_id: string;
  chapter_title: string;
  chapter_path: string;
  chapter_file_type: "dir" | "cbz";
  last_read_at: number;
}
